sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -55,6 +60,11 @@ public BufferHolder(UnsafeRow row, int initialSize) {
    * Grows the buffer by at least neededSize and points the row to the buffer.
    */
   public void grow(int neededSize) {
+    if (neededSize > Integer.MAX_VALUE / 2 - totalSize()) {
```

Comment:
shall we move this check into the if branch below? then we can just check `length * 2 <= Integer.MAX_VALUE` and others can understand it very easily as there is a `final byte[] tmp = new byte[length * 2];` next line.